ambari-server/src/main/java/org/apache/ambari/server/checks/KafkaPropertiesCheck.java

```diff
@@ -0,0 +1,113 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.checks;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.ambari.server.AmbariException;
+import org.apache.ambari.server.controller.PrereqCheckRequest;
+import org.apache.ambari.server.state.Cluster;
+import org.apache.ambari.server.state.ServiceInfo;
+import org.apache.ambari.server.state.stack.PrereqCheckStatus;
+import org.apache.ambari.server.state.stack.PrerequisiteCheck;
+import org.apache.ambari.server.state.stack.upgrade.UpgradeType;
+
+import com.google.inject.Singleton;
+
+/**
+ * Check Kafka configuration properties before upgrade.
+ *
+ * Property to check:
+ *  inter.broker.protocol.version
+ *  log.message.format.version
+ *
+ * These configurations must exist and have a value (the value for both should be aligned with the Kafka version on the current stack).
+ *
+ * For inter.broker.protocol.version :
+ * value is not empty
+ * value is set to current Kafka version in the stack (e.g. for HDP 2.6.x value should be 0.10.1, HDP 2.5.x should be 0.10.0, HDP 2.3x - 2.4x should be 0.9.0)
+ *
+ * For log.message.format.version:
+ * value is not empty (version can vary from current stack version)
+ */
+@Singleton
+@UpgradeCheck(
+    group = UpgradeCheckGroup.REPOSITORY_VERSION,
+    required = { UpgradeType.ROLLING, UpgradeType.NON_ROLLING, UpgradeType.HOST_ORDERED })
+public class KafkaPropertiesCheck extends AbstractCheckDescriptor {
+  private static String KAFKA_BROKER_CONFIG = "kafka-broker";
+  private static String KAFKA_SERVICE_NAME = "KAFKA";
+
+  private interface KafkaProperties{
+    String INTER_BROKER_PROTOKOL_VERSION = "inter.broker.protocol.version";
+    String LOG_MESSAGE_FORMAT_VERSION = "log.message.format.version";
+    List<String> ALL_PROPERTIES = Arrays.asList(INTER_BROKER_PROTOKOL_VERSION, LOG_MESSAGE_FORMAT_VERSION);
+  }
+
+  /**
+   * Constructor.
+   */
+  public KafkaPropertiesCheck() {
+    super(CheckDescription.KAFKA_PROPERTIES_VALIDATION);
+  }
+
+  private String getKafkaServiceVersion(Cluster cluster)throws AmbariException{
+    ServiceInfo serviceInfo = ambariMetaInfo.get().getStack(cluster.getCurrentStackVersion()).getService(KAFKA_SERVICE_NAME);
```

Comment:
this is the basic place where we can get a version, however we have a second one `repositoryVersionHelper.get().getRepositoryVersionEntity(cluster, cluster.getService("KAFKA").getServiceComponent("KAFKA_BROKER")).getRepositoryXml().manifests`, which feels better. I choosed first one as the change apply 2.3+ stacks, which may not have any vdf (not sure)

Reply:
But what about those versions that DO have a VDF?  That would be the more accurate check, no?  We should also try to validate that that version is one of the "known" ones.  What if the vdf/stack defined some 0.10.1.2 or something?

Reply:
We have no way to validate component version. Ambari not support component version retrieval, only stack version.   And internally component version in ambari = stack component version. Since this, the only source for the validation of "known" version would be meta info and vdf, where vdf can't be trusted as it is user-defined and may have no relation to actually supported versions in the checked properties.

Meta information, is the source supported from the very beginning and any custom change to internal resources is a responsibility of the one who changed it. 

In summary i prefer to follow jira description versions, and the ones mentioned in Kafka documentation.

Reply:
Ok. I figured I'd bring it up anyway.  Thanks!